extensions/hadoop/src/main/java/com/hazelcast/jet/hadoop/impl/JsonInputFormat.java

```diff
@@ -30,68 +32,170 @@
 import org.apache.hadoop.mapreduce.RecordReader;
 import org.apache.hadoop.mapreduce.TaskAttemptContext;
 import org.apache.hadoop.mapreduce.lib.input.FileInputFormat;
+import org.apache.hadoop.mapreduce.lib.input.FileSplit;
 import org.apache.hadoop.mapreduce.lib.input.LineRecordReader;
 
 import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.Iterator;
 import java.util.function.Function;
 
-public class JsonInputFormat extends FileInputFormat<LongWritable, Object> {
+import static java.util.Collections.emptyIterator;
+
+public class JsonInputFormat extends FileInputFormat<NullWritable, Object> {
 
     public static final String JSON_INPUT_FORMAT_BEAN_CLASS = "json.bean.class";
+    public static final String JSON_MULTILINE = "json.multiline";
```

Comment:
To support parallel reading we would need to reimplement the logic handling input splits. It would include somehow wrapping the iterator returned from `JsonUtil.mapSequenceFrom` to stop when we reach the input split boundary.

I think this is easier, the `SingleLineJsonRecordReader` mostly delegates to LineRecordReader.